### DISCUSSION TITLE
Feature Request - Option to play stored sequence on start up

### DISCUSSION THREAD

**Discussion by lathemaster** [forkineye/ESPixelStick#557] [Ideas]
Really like what I am seeing in V4 Beta4
For stand alone displays I would like to play a selected sequence on powerup

**Comment by brmeadows2841** [comment]
If you have an SD card, upload your sequences to the SD card. If using Xlights, via tools/FPP connect.

Look under device setup, change "FSEQ File To Play" to the sequence you want to play.

**Reply by lathemaster** [reply]
So if I have a sequence or playlist selected and saved. That sequence will play when the device powers up. I'll give it a try
Thanks

**Comment by brmeadows2841** [comment]
Change "FSEQ File To Play" to the sequence you want to play. It should play on reboot.

If it receives data from a source (FPP/Xlights) it will stop playing the sequence and follow the Data stream from the sending source.

One problem you will have is : when ESPixelstick is set this way Xlights /FPP connect will no longer see the ESPixelstick device. In other words, the setting has to be changed back to Play remote Sequence in order to upload any changes using FPP connect.

I don't think the present ESPixelstick has a playlist feature, would someone official correct me if I'm wrong.

I know there is code for a playlist built into ESPixelstick but I don't know how to use it.

A playlist feature would be awesome.

**Reply by lathemaster** [reply]
Thanks - Yes I found out that I had to stop the sequence playing to upload anther but that is OK for thei application.
This is my 1st year doinf a total pixel Christmas display. I have an FPP controller to put together but in case I run out of time the ESPixelSticks are my backup. Not going to do anything crazy for the 1st year. 9 3D printed snowflakes, 12 wrapped tree trunks and hopefully a Boscoyo wreath above the frontdoor. A few simple sequences and no music.

**Comment by brmeadows2841** [comment]
I would setup an FPP controller first, it can control your show/schedules by time /date etc.
Set the ESPixelSticks as FPP remote, then upload to your devices.
Turn on multi-sync in FPP it will tell the remote devices when/what to play.
I would image FPP 5.5 to your PI or Beaglebone.

![FPP-v5 5-BBB](https://user-images.githubusercontent.com/81775646/187948395-db7db154-d75c-4ee1-8d68-3308213c7825.png)

**Reply by lathemaster** [reply]
Thanks - all hinges on me getting the FPP controller built

**Comment by MartinMueller2003** [comment]
Yes V4 has the ability to play a list of sequences and effects. It also supports replay counts so you an play a sequence N times and then move to another in the list. The playlist will repeat when it reaches the end. The format for the playlist is in the documentation.

**Comment by MartinMueller2003** [comment]
You do not need FPP if you are going to have V4 play from a file list. You only need FPP if you are looking for an inexpensive show player that does not tie up a computer while your show is playing.

**Reply by lathemaster** [reply]
Got it thanks

**Comment by MartinMueller2003** [comment]
But this conversation brings up a new feature:

Play a play list or preconfigured file until E1.31/DDP/FPP Sync is received. Only E1.31 or DDP would stop the playlist today.

**Comment by brmeadows2841** [comment]
I agree with Martin on most of this. The ESPixelStick software is very capable. If I remember correctly ESPixelStick installed on an ESP32 is capable of 680 pixels per port. With the current version of the software on an ESP32, it can have 6 ports, depends on how many GIPOs are available.

The reason I said" setup FPP", it could take the place of a computer, and it sounded like you had multiple ESPixelSticks.
FPP could output the sequence data to multiple devices, ESPixelSticks, or other FPP devices.

Martin, is there any way for ESPixelSticks to communicate with each other? What I'm asking is there a way to setup ESPixelSticks in a Master Slave relationship?

**Reply by lathemaster** [reply]
Yes - multiple ESPixelSticks controlling a "cluster" of props is my plan B in case I don't get my FPP controller built and sequenced in time. I am using the ESPixelSticks for testing completed props and learning some single sequencing

**Reply by MartinMueller2003** [reply]
There is no 680 per port limit. There is a limit of 3000 pixels on ESP32 devices that do not have PSRAM and 7000 pixels on systems that do have PSRAM. The distribution is completely up to the user. If you want a 25ms refresh rate then you need to stay under (or at) 680 pixels on a port. ESP8266 has a 1200 pixel limit.

**Reply by MartinMueller2003** [reply]
Also, multiple ESP V4 devices do not coordinate between each other. A cluster of ESP V4s would require an FPP Master or an xLights show player running in FPP Master mode.

**Reply by MartinMueller2003** [reply]
If you are asking for a new feature in which one of the ESP V4s pretends to be an FPP Master, yes that could be added. However, based on my schedule, that would be started in February. If you want this, please put in a feature request.

**Reply by brmeadows2841** [reply]
> If you are asking for a new feature in which one of the ESP V4s pretends to be an FPP Master, yes that could be added. However, based on my schedule, that would be started in February. If you want this, please put in a feature request.

Thanks for all you do.

I summitted a feature request per this conversation.

#559

**Comment by brmeadows2841** [comment]
Martin you mentioned the Playlist in the documentation.

Is this the correct format?
[

 { 
         "type": "file",
        "name": "AllAboard.fseq",
        "playcount": 1
    },

    {
        "type": "file",
        "name": "Firefly.fseq",
        "playcount": 1
    },

    {
        "type": "file",
        "name": "01 - Carol of the Bells.fseq",
        "playcount": 1

},
   
    {
        "type": "pause",
        "duration": 2
}
  
]

**Reply by MartinMueller2003** [reply]
Yes that is the format

**Reply by MartinMueller2003** [reply]
Hmm. I thought the example also included using effects. Did you cut those out?

**Reply by brmeadows2841** [reply]
Yes, I cutout everything but the file segment to keep things simple.

**Comment by paulu10** [comment]
Firmware version - 4.0-ci19209534132 
Device - ESP32 
@MartinMueller2003 I am having an issue where individual fseq files can be played from the SD card without any issues, but when a playlist is created with the same file names only the first fseq file plays in a loop.

 [
    {
      "type": "file",
      "name": "file1.fseq",
      "playcount": 1
    },
    {
      "type": "file",
      "name": "file2.fseq",
      "playcount": 1
    },
 {
      "type": "file",
      "name": "file3.fseq",
      "playcount": 1
    },
    {
      "type": "pause",
      "duration": 5
    }
  ]

**Comment by MartinMueller2003** [comment]
Please create a defect issue. This conversation is closed. I will take a look at this ASAP.
